service/src/main/java/com/codeforcommunity/processor/TeamsProcessorImpl.java

```diff
@@ -280,4 +295,47 @@ public GetUserTeamsResponse getUserTeams(JWTData userdata) {
     }
     return teamMembers;
   }
+
+  @Override
+  public void transferOwnership(JWTData userData, TransferOwnershipRequest request) {
+    int ownerId = userData.getUserId();
+
+    UserTeamRecord userTeamRecordOwner =
+        db.selectFrom(USER_TEAM)
+            .where(USER_TEAM.USER_ID.eq(ownerId))
+            .and(USER_TEAM.TEAM_ID.eq(request.getTeamId()))
+            .fetchOneInto(UserTeamRecord.class);
+
+    if (userTeamRecordOwner == null || userTeamRecordOwner.getTeamRole() != TeamRole.LEADER) {
+      throw new TeamLeaderOnlyRouteException(request.getTeamId());
+    }
+
+    Optional<UsersRecord> newOwnerOptional =
+        db.selectFrom(USERS).where(USERS.EMAIL.eq(request.getNewOwnerEmail())).fetchOptional();
+
+    UsersRecord newOwner =
+        newOwnerOptional.orElseThrow(
+            () -> new UserDoesNotExistException(request.getNewOwnerEmail()));
+
+    boolean userOnTeam =
+        db.fetchExists(
+            db.selectFrom(USER_TEAM)
+                .where(USER_TEAM.USER_ID.eq(newOwner.getId()))
+                .and(USER_TEAM.TEAM_ID.eq(request.getTeamId())));
+
+    if (!userOnTeam) {
+      throw new UserNotOnTeamException(newOwner.getId(), request.getTeamId());
+    }
+
+    db.update(USER_TEAM)
```

Comment:
Why not just use the record we already have?

Reply:
Jooq isn't generating `UpdatableRecordImpl`s for that table since it doesn't have any primary keys, so it can't be updated like that. I could go in an add a compound key on the `user_id and `team_id` columns, and then I could use that. But otherwise, this is the only way to update it. 

[UpdatableRecordImpl documentation](https://www.jooq.org/javadoc/latest/org.jooq/org/jooq/UpdatableRecord.html)